src/python/pants/net/http/fetcher.py

```diff
@@ -176,24 +179,26 @@ def status(self, code, content_length=None):
       self.read = 0
 
     def recv_chunk(self, data):
+      data = [str(num).encode('utf-8') for num in data] if isinstance(data, list) else str(data).encode('utf-8')
```

Comment:
Numbers must first be cast to strings so that encode can work. Not calling `bytes()` directly, because its behavior changes in Py2 vs Py3, i.e. `b'1'` vs `b'\x00'`